pkg/operator/controller/ingress/status.go

```diff
@@ -77,6 +79,54 @@ func mergeConditions(conditions []operatorv1.OperatorCondition, updates ...opera
 	return conditions
 }
 
+// computeIngressTLSProfile computes the ingresscontroller's current TLS
+// profile.  If the deployment is ready, then the TLS profile is inferred from
+// deployment's pod template spec.  Otherwise the previous TLS profile is used.
+func computeIngressTLSProfile(oldProfile *configv1.TLSProfileSpec, deployment *appsv1.Deployment) *configv1.TLSProfileSpec {
+	if deployment.Status.Replicas != deployment.Status.UpdatedReplicas {
+		return oldProfile
+	}
+
+	containers := deployment.Spec.Template.Spec.Containers
+	if len(containers) < 1 || containers[0].Name != "router" {
+		return &configv1.TLSProfileSpec{}
+	}
+
+	var (
+		ciphersString       string
+		minTLSVersionString string
+	)
+	for _, envVar := range containers[0].Env {
```

Comment:
This seems to be an abstraction leak from the controller code that knows how to create the router deployment from an ingresscontroller. For example, if we ever changed the way that we configure the router implementation's ciphers, does this become a problem? Let's say we change the env var key. This code now has to be updated and backwards compatible. What if we change to an haproxy API call? Would we change this code to query haproxy?

Reply:
> This seems to be an abstraction leak from the controller code that knows how to create the router deployment from an ingresscontroller.

It's all in the same package.

> For example, if we ever changed the way that we configure the router implementation's ciphers, does this become a problem? Let's say we change the env var key. This code now has to be updated and backwards compatible.

Again, this is all in the same package, same controller.  This change does not couple `pkg/operator/controller/ingress/status.go` any more tightly to HAProxy than `pkg/operator/controller/ingress/deployment.go` is already coupled to HAProxy.  Would moving this logic (inferring a TLS profile spec from a deployment pod template spec) to `pkg/operator/controller/ingress/deployment.go` resolve your concern?

> What if we change to an haproxy API call? Would we change this code to query haproxy?

I hadn't considered that the operator itself might make API call to HAProxy, and I can only imagine what form the communication would take.  I speculate we would define a networkpolicy to grant the operator access to the operand pods, and the operator would attempt to update all HAProxy instances via their podips and then retry on failure or update the status synchronously on success.

Reply:
Okay, thanks.

>Would moving this logic (inferring a TLS profile spec from a deployment pod template spec) to pkg/operator/controller/ingress/deployment.go resolve your concern?

This might be good, but it can be a followup refactoring. I wouldn't block the PR on it.